### ISSUE TITLE
Services should be able to create a functioning external load balancer is failing on Jenkins

### ISSUE THREAD

**Issue by piosz** [GoogleCloudPlatform/kubernetes#5846]
```
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:338
Expected error:
    <*errors.StatusError | 0xc2081b8680>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {SelfLink: "", ResourceVersion: ""},
            Status: "Failure",
            Message: "The resource 'projects/kubernetes-jenkins/regions/us-central1/targetPools/e2e-test-jenkins-default-external-lb-test' already exists",
            Reason: "",
            Details: nil,
            Code: 500,
        },
    }
    The resource 'projects/kubernetes-jenkins/regions/us-central1/targetPools/e2e-test-jenkins-default-external-lb-test' already exists
not to have occurred
```

**Comment by zmerlynn** [comment]
It's because it failed once, is now leaked, and will never recover without
external intervention. cc @quinton-hoole

**Comment by piosz** [comment]
I know, but maybe we can avoid such situation in the future.

**Comment by quinton-hoole** [comment]
The best type of solution we have right now is the one that zmerlynn@ implemented in PR #5732, i.e. clean up external resources as part of the pre-checks on the next invocation of the test.  I plan to come up with a more generic solution based on projects or namespaces, but that will probably take a while still.  In the mean time, please implement a cleanup as per PR #5732.

**Comment by zmerlynn** [comment]
Read https://github.com/GoogleCloudPlatform/kubernetes/issues/5722#issuecomment-84446754: That cleanup isn't enough for Jenkins, because it's trying to cleanup the next run. (@a-robinson actually implemented a cleanup like this, but it only works for the ^C case.)

Two things here: (a) the controller could reclaim/update ELBs that are around with the same name (or not), and (b) we need a resource hoover pre-/post-condition test for Jenkins.

**Comment by a-robinson** [comment]
I think when Quinton said "clean up external resources", he meant the external load balancer directly. It's unfortunate that this would have to be cloud-platform specific.

I'm working on making load balancer creation asynchronous to service creation. This won't directly fix the problem, but would make it easier to add in logic for handling ELBs that already exist.

In the meantime, perhaps the best option is just to increase the timeouts.

**Comment by quinton-hoole** [comment]
I'm going to make it run in a timestamped namespace, as per #6125.

**Comment by quinton-hoole** [comment]
Resolved by #6847 and #6271.  Closing.

**Comment by bprashanth** [comment]
For the record I just deleted a bunch of targetpools and static ips that were preventing gce-flaky clusters from being born (lack of static ip quota).
